vertical-pod-autoscaler/pkg/utils/vpa/capping.go

```diff
@@ -153,6 +156,30 @@ func applyVPAPolicy(recommendation apiv1.ResourceList, policy *vpa_types.Contain
 	return annotations
 }
 
+// ApplyVPAPolicy returns a recommendation, adjusted to obey policy.
+func ApplyVPAPolicy(podRecommendation *vpa_types.RecommendedPodResources,
+	policy *vpa_types.PodResourcePolicy) (*vpa_types.RecommendedPodResources, error) {
+	if podRecommendation == nil {
+		return nil, nil
+	}
+	if policy == nil {
+		return podRecommendation, nil
+	}
+
+	updatedRecommendations := []vpa_types.RecommendedContainerResources{}
+	for _, containerRecommendation := range podRecommendation.ContainerRecommendations {
+		containerName := containerRecommendation.ContainerName
+		updatedContainerResources, _, err := getCappedRecommendationForContainer(
```

Comment:
Can you please double-check if reusing the same function while ignoring the computed annotations is performant?

I recall a story that somewhere in CA creation of error messages which were later ignored proved to be the biggest performance issue around simulations. There it has been solved by adding an argument like "silent/verbose".

Reply:
Went with not reusing.